### PR TITLE
Prevent system with items being deleted #127

### LINKED ISSUE #127: Add items check when deleting systems
Should not be able to delete a system that contains items. Ensure front end also displays the right message.

### PR DESCRIPTION
**Pull request by joelvdavies**
## Description
Adds check to ensure a system doesn't have any items before it's deleted.

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
Closes #127

### CODE REVIEW

File: inventory_management_system_api/repositories/system.py [hunk 1]
```diff
@@ -171,12 +172,14 @@ def _is_duplicate_system(self, parent_id: Optional[str], code: str) -> bool:
...
         """
         logger.info("Checking if system with ID '%s' has child elements", str(system_id))
-        # Check if it has System's
-        system = self._systems_collection.find_one({"parent_id": system_id})
-        return system is not None
+
+        return (
+            self._systems_collection.find_one({"parent_id": system_id}) is not None
+            or self._items_collection.find_one({"system_id": system_id}) is not None
+        )
```
Comment (VKTB) [inline inventory_management_system_api/repositories/system.py:185]: I like this as it won't execute the second query if the first returns. We should do the same for catalogue category.

File: test/e2e/test_manufacturer.py [hunk 2]
```diff
@@ -3,34 +3,33 @@
...
     manufacturer = response.json()
 
-    assert manufacturer["name"] == manufacturer_post["name"]
-    assert manufacturer["url"] == manufacturer_post["url"]
-    assert manufacturer["address"] == manufacturer_post["address"]
-    assert manufacturer["telephone"] == manufacturer_post["telephone"]
+    assert manufacturer["name"] == MANUFACTURER_POST["name"]
+    assert manufacturer["url"] == MANUFACTURER_POST["url"]
+    assert manufacturer["address"] == MANUFACTURER_POST["address"]
+    assert manufacturer["telephone"] == MANUFACTURER_POST["telephone"]
```
Comment (VKTB) [inline test/e2e/test_manufacturer.py:32]: I know you only modified this piece of code but shouldn't it be asserting the whole object?

File: test/e2e/test_system.py [hunk 3]
```diff
@@ -542,3 +545,38 @@ def test_delete_system_with_child_system(test_client):
 
     assert response.status_code == 409
     assert response.json()["detail"] == "System has child elements and cannot be deleted"
+
+
+def test_delete_system_with_child_item(test_client):
```
Comment (VKTB) [inline test/e2e/test_system.py:550]: Can we have another test with child system?

Comment (joelvdavies) [reply test/e2e/test_system.py:550]: That one is above (it already existed)

File: test/e2e/test_manufacturer.py (hunk 2 above)
Comment (joelvdavies) [reply test/e2e/test_manufacturer.py:32]: Yeah I though it best to leave it for #90 as the rest of the file is like that.

File: test/e2e/test_system.py (hunk 3 above)
Comment (VKTB) [reply test/e2e/test_system.py:550]: Ah my bad, sorry.

File: inventory_management_system_api/repositories/system.py (hunk 1 above)
Comment (VKTB) [reply inventory_management_system_api/repositories/system.py:185]: Made a comment about this in https://github.com/ral-facilities/inventory-management-system-api/pull/148#discussion_r1450366862

**Review by VKTB — APPROVED** [review]

**Review by VKTB — APPROVED** [review]

**Pull request merged**